### PR TITLE
LL-6567 - SWAP - Tracking Events

### PR DESCRIPTION
**Pull request by jelbaz-ledger**
## 🦒 Context (issues, jira)

Trying to stick to the old events name, with an additional `swapVersion` property to disambiguate.

Also:
- fixed: the swapId prop in the recap that was mocked 🤦 
- fixed: preselect the right account when coming from the accounts page

[LL-6567]

[Events Reference](https://docs.google.com/spreadsheets/d/1ECRYMrOjHh76lgtfBiY4vc5Q4Z3eu-LwAF7SiSWgwqU/edit#gid=1168302553)

## 💻  Description / Demo (image or video)

*N/A*

## 🖤  Expectations to reach

PR must pass CI, merge develop if conflicts, do not force push. Thanks!

- **on QA**: at least one of these two checkboxes must be checked:
  - [ ] a specific test planned is defined on Jira
  - [ ] this PR is covered by automatic UI test
- **on delivery**: at least one of these two checkboxes must be checked: 
  - [x] Option 1: **no impact**: The changes of this PR have ZERO impact on the userland (invisible for users)
  - [ ] Option 2: **atomic delivery**: the changes is atomic and complete (no partial delivery)

PR must pass CI, merge develop if conflicts, do not force push. Thanks!

[LL-6567]: https://ledgerhq.atlassian.net/browse/LL-6567

### CODE REVIEW

File: src/renderer/screens/exchange/Swap2/utils/shared/useSwapTransaction.js [hunk 1]
```diff
@@ -77,15 +78,31 @@ export type SwapTransactionType = {
...
+  defaultParentAccount,
+}: {
+  defaultCurrency?: $PropertyType<SwapSelectorStateType, "currency">,
+  defaultAccount?: $PropertyType<SwapSelectorStateType, "account">,
+  defaultParentAccount?: $PropertyType<SwapSelectorStateType, "parentAccount">,
+} = {}): SwapTransactionType => {
   const [toState, setToState] = useState<SwapSelectorStateType>(SelectorStateDefaultValues);
-  const [fromState, setFromState] = useState<SwapSelectorStateType>(SelectorStateDefaultValues);
+  const [fromState, setFromState] = useState<SwapSelectorStateType>({
+    ...SelectorStateDefaultValues,
```
Comment (juan-cortes) [inline src/renderer/screens/exchange/Swap2/utils/shared/useSwapTransaction.js:92]: Lowercase, it's not a class nor a component

**Review by juan-cortes — CHANGES_REQUESTED** [review]

File: src/renderer/screens/exchange/Swap2/Form/FeesDrawer/index.js [hunk 2]
```diff
@@ -23,6 +30,13 @@ export default function FeesDrawer({ swapTransaction, disableSlowStrategy = fals
 
   return (
     <Box height="100%">
+      <TrackPage
+        category="Swap"
+        name="Form - Edit Fees"
```
[the remark this answers is not in the document]
Comment (jelbaz-ledger) [reply src/renderer/screens/exchange/Swap2/Form/FeesDrawer/index.js:35]: I simply reused the names from the reference spreadsheet, I don't think it's my call to name these events :/

File: src/renderer/screens/exchange/Swap2/utils/index.js [hunk 3]
```diff
@@ -3,6 +3,11 @@ import { useCallback } from "react";
 import { useHistory } from "react-router-dom";
 import * as providerIcons from "~/renderer/icons/providers";
 import type { ExchangeRate } from "@ledgerhq/live-common/lib/exchange/swap/types";
+import { SwapExchangeRateAmountTooLow } from "@ledgerhq/live-common/lib/errors";
+import { NotEnoughBalance } from "@ledgerhq/errors";
+import { track } from "~/renderer/analytics/segment";
+
+export const SWAP_VERSION = "2.34";
```
[the remark this answers is not in the document]
Comment (jelbaz-ledger) [reply src/renderer/screens/exchange/Swap2/utils/index.js:10]: To disambiguate events having the same name as the current swap. Again, I took the value from the [spreadsheet](https://docs.google.com/spreadsheets/d/1ECRYMrOjHh76lgtfBiY4vc5Q4Z3eu-LwAF7SiSWgwqU/edit#gid=1168302553).

File: src/renderer/screens/exchange/Swap2/utils/shared/useSwapTransaction.js (hunk 1 above)
Comment (jelbaz-ledger) [reply src/renderer/screens/exchange/Swap2/utils/shared/useSwapTransaction.js:92]: You mean for `SelectorStateDefaultValues` ? I just reused the var here, but yeah I can find/replace everywhere.

File: src/renderer/screens/exchange/Swap2/Form/FeesDrawer/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (juan-cortes) [reply src/renderer/screens/exchange/Swap2/Form/FeesDrawer/index.js:35]: Wasn't blaming you for the inconsistency, but sure, just ignore it then

File: src/renderer/screens/exchange/Swap2/utils/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (juan-cortes) [reply src/renderer/screens/exchange/Swap2/utils/index.js:10]: 🤦🏼  ok

File: src/renderer/screens/exchange/Swap2/utils/shared/useSwapTransaction.js (hunk 1 above)
Comment (juan-cortes) [reply src/renderer/screens/exchange/Swap2/utils/shared/useSwapTransaction.js:92]: 👍🏼

**Comment by jelbaz-ledger** [comment]
@juan-cortes I'm merging the PR since we need to advance on that front (and the refactor/cleanup PR depends on this one). If you have any question/remarks regarding events I think that you can check with Chris directly. Then if there are changes I'll commit them at a later time ⏳

**Pull request merged**